container-accesslogging/src/main/java/com/yahoo/container/logging/LogFileHandler.java

```diff
@@ -263,15 +264,28 @@ private void internalRotateNow() throws InterruptedException {
         numberOfRecords = 0;
         lastRotationTime = now;
         nextRotationTime = 0; //figure it out later (lazy evaluation)
-        if (compressOnRotation && (oldFileName != null)) {
-            triggerCompression(oldFileName);
+        if ((oldFileName != null)) {
+            if (compressOnRotation) {
+                triggerCompression(oldFileName);
+            } else {
+                NativeIO nativeIO = new NativeIO();
+                nativeIO.dropFileFromCache(new File(oldFileName));
+            }
         }
     }
 
     private void triggerCompression(String oldFileName) {
         try {
+            String zippedFileName = oldFileName + ".gz";
             Runtime r = Runtime.getRuntime();
-            Process p = r.exec(new String[] { "gzip", oldFileName });
+            StringBuilder cmd = new StringBuilder("gzip");
+            cmd.append(" < "). append(oldFileName).append(" > ").append(zippedFileName);
+            Process p = r.exec(cmd.toString());
+            NativeIO nativeIO = new NativeIO();
+            File oldFile = new File(oldFileName);
+            nativeIO.dropFileFromCache(oldFile);
```

Comment:
why are deleted files not dropped from the cache automatically?

Reply:
In case there are other references to it preventing evaporating immediately.
Probably not strictly necessary, but a precaution.